### ISSUE TITLE
Cannot find buffer binding for index 0

### ISSUE THREAD

**Issue by CuriousMike56** [RigsOfRods/rigs-of-rods#2024]
I've been experiencing this issue for a little while now, decided I'd create an issue on it since it happened again and I have a log of the crash now.

The crash seems to be random, It's happened multiple times attempting to open the vehicle selector right after loading a terrain, once while driving around Penguinville, and now right after spawning a vehicle on Simple Terrain:
![ror_2019-01-03_19-39-45](https://user-images.githubusercontent.com/46073351/50669244-07ae0e00-0f92-11e9-88e1-e6456887b445.png)
The window loses focus right before the crash. 
These are the last lines in the log:
```
19:39:29: D3D9 Device 0x[00000280F7973980] entered lost state
19:39:29: D3D9 device: 0x[00000280F7973980] lost. Releasing D3D9 texture: Refraction
19:39:29: Released D3D9 texture: Refraction
19:39:29: D3D9 device: 0x[00000280F7973980] lost. Releasing D3D9 texture: EnvironmentTexture
19:39:29: Released D3D9 texture: EnvironmentTexture
19:39:29: D3D9 device: 0x[00000280F7973980] lost. Releasing D3D9 texture: Reflection
19:39:29: Released D3D9 texture: Reflection
19:39:29: D3D9 device: 0x[00000280F7973980] lost. Releasing D3D9 texture: RTTTexture1
19:39:29: Released D3D9 texture: RTTTexture1
19:39:29: D3D9 device: 0x[00000280F7973980] lost. Releasing D3D9 texture: MapRttTex-1
19:39:29: Released D3D9 texture: MapRttTex-1
19:39:29: D3D9 device: 0x[00000280F7973980] lost. Releasing D3D9 texture: SceneManagerInstance1/compRTT
19:39:29: Released D3D9 texture: SceneManagerInstance1/compRTT
19:39:29: D3D9 device: 0x[00000280F7973980] lost. Releasing D3D9 texture: MapRttTex-2
19:39:29: Released D3D9 texture: MapRttTex-2
19:39:29: D3D9 device: 0x[00000280F7973980] lost. Releasing D3D9 texture: dashtexture
19:39:29: Released D3D9 texture: dashtexture
19:39:29: D3D9 Device 0x[00000280F7973980] entered lost state
19:39:29: Ogre::ItemIdentityException::ItemIdentityException: Cannot find buffer binding for index 0 in VertexBufferBinding::unsetBinding at c:\conan\.conan\data\ogre\1.11.5\anotherfoxguy\stable\build\2d13f0c58bafe8a9fae74369aea9c086e652e9f3\ogremain\src\ogrehardwarevertexbuffer.cpp (line 769)
```
Will upload the full log if required. Running windowed mode with input grabbing set to dynamic if that helps any. I also have ReShade installed, though I don't think that would cause it as no one else seems to have this issue.

**Comment by tritonas00** [comment]
it happens with opengl also?

**Comment by CuriousMike56** [comment]
No idea, I can't easily reproduce the crash.

**Comment by tritonas00** [comment]
https://forums.ogre3d.org/viewtopic.php?t=94697

**Comment by CuriousMike56** [comment]
Okay, just figured out I can reproduce the crash every time by using full screen and pressing CTRL+ALT+DEL after loading a terrain.

**Comment by tritonas00** [comment]
this is now fixed with https://github.com/RigsOfRods/rigs-of-rods/commit/d31c5e50b340177d944fae75a3960b7f2d77f6ce?

**Comment by CuriousMike56** [comment]
No, that commit only fixed the texture tool.

**Comment by tritonas00** [comment]
@paroj can you help on this?

**Comment by paroj** [comment]
probably related to context loss on alt+tabbing. Try random combinations of "D3D9Ex" and "Resource Creation Policy". Unfortunately I dont know much about how D3D9 works.

**Comment by paroj** [comment]
might be this https://github.com/OGRECave/ogre/pull/1084/commits/766902e6908636aaf44cde8baef30d258e353f9c

**Comment by CuriousMike56** [comment]
Any update on this? Loading the latest Community Map throws the same error:
```
17:36:26: D3D9 device: 0x[000002A1094DCB00] was reset
17:36:45: D3D9 Device 0x[000002A1094DCB00] entered lost state
17:36:45: D3D9 device: 0x[000002A1094DCB00] lost. Releasing D3D9 texture: RTTTexture1
17:36:45: Released D3D9 texture: RTTTexture1
17:36:45: D3D9 device: 0x[000002A1094DCB00] lost. Releasing D3D9 texture: EnvironmentTexture
17:36:45: Released D3D9 texture: EnvironmentTexture
17:36:45: D3D9 device: 0x[000002A1094DCB00] lost. Releasing D3D9 texture: Refraction
17:36:45: Released D3D9 texture: Refraction
17:36:45: D3D9 device: 0x[000002A1094DCB00] lost. Releasing D3D9 texture: Reflection
17:36:45: Released D3D9 texture: Reflection
17:36:45: D3D9 device: 0x[000002A1094DCB00] lost. Releasing D3D9 texture: dashtexture
17:36:45: Released D3D9 texture: dashtexture
17:36:45: D3D9 device: 0x[000002A1094DCB00] lost. Releasing D3D9 texture: MapRttTex-1
17:36:45: Released D3D9 texture: MapRttTex-1
17:36:45: D3D9 device: 0x[000002A1094DCB00] lost. Releasing D3D9 texture: SceneManagerInstance1/compRTT
17:36:45: Released D3D9 texture: SceneManagerInstance1/compRTT
17:36:45: D3D9 device: 0x[000002A1094DCB00] lost. Releasing D3D9 texture: 630ech_camera_0@Actor_0_texture
17:36:45: Released D3D9 texture: 630ech_camera_0@Actor_0_texture
17:36:45: D3D9 device: 0x[000002A1094DCB00] lost. Releasing D3D9 texture: MapRttTex-2
17:36:45: Released D3D9 texture: MapRttTex-2
17:36:45: D3D9 Device 0x[000002A1094DCB00] entered lost state
17:36:45: Ogre::ItemIdentityException::ItemIdentityException: Cannot find buffer binding for index 0 in VertexBufferBinding::unsetBinding at c:\conan\.conan\data\ogre\1.11.5\anotherfoxguy\stable\build\2d13f0c58bafe8a9fae74369aea9c086e652e9f3\ogremain\src\ogrehardwarevertexbuffer.cpp (line 769)
```

**Comment by CuriousMike56** [comment]
Appears to be fixed now with 1.11.6. 👍
